handlers/src/main/java/com/nextdoor/bender/handler/kinesis/KinesisHandler.java

```diff
@@ -70,4 +77,27 @@ public void onException(Exception e) {
   public InternalEventIterator<InternalEvent> getInternalEventIterator() {
     return this.recordIterator;
   }
+
+  @Override
+  public void prepareMetadata() {
+    /**
+     * Instantiate the HandlerMetadata object
+     */
+    super.prepareMetadata();
```

Comment:
I don't like doing it this way because there is nothing that forces `super.prepareMetadata();` being called. Instead:

Create a `abstract HandlerMetadata getHandlerMetadata();` in `BaseHandler`. Then in `BaseHandler.processInternal()` method call `getHandlerMetadata()` and add any additional fields like you do below with `metadata.setField(...)`.